### PR TITLE
swarm introspection: review comments

### PR DESCRIPTION
**Pull request by raulk**
This PR simplifies the introspection code. It mostly removes the DTO/entity Javabean reminiscent-like approach to access fields. I've also reviewed the sourcing of all fields, wired those that I could, and added 

There was a bug with traffic metrics sourcing in your initial PR. The Bandwidth Reporter is a swarm-wide object, and does not keep track of metrics at the stream level. Your initial PR, you were mixing up traffic from all streams.

I've made the ID fields on the connection and the stream an uint32, sourced from an atomic counter at the swarm-level. It does not guarantee uniqueness across restarts, of course, but there are cheaper ways we can achieve canonicality across restarts (e.g. a UUID for the runtime).

### CODE REVIEW

File: introspect.go [hunk 1]
```diff
@@ -1,173 +1,219 @@
...
-func (swarm *Swarm) IntrospectConnections(query introspect.ConnectionQueryParams) ([]*introspectpb.Connection, error) {
-	swarm.conns.RLock()
-	defer swarm.conns.RUnlock()
-
-	containsId := func(ids []introspect.ConnectionID, id string) bool {
-		for _, i := range ids {
-			if string(i) == id {
-				return true
+func (s *Swarm) IntrospectConnections(q introspection.ConnectionQueryParams) ([]*introspection_pb.Connection, error) {
+	var conns []network.Conn
```
Comment (aarshkshah1992) [inline introspect.go:36]: This is nice. Re-uses the existing lock/iteration code in Swarm.

File: introspect.go [hunk 2]
```diff
@@ -1,173 +1,219 @@
...
-			is, err := swarm.introspectStream(s)
-			if err != nil {
-				return errors.Wrap(err, "failed to convert stream to introspect.Stream")
-			}
-			sl.Streams = append(sl.Streams, is)
-		} else {
-			sl.StreamIds = append(sl.StreamIds, []byte(s.id))
-		}
-		return nil
+	// TODO can we get the transport ID from the multiaddr?
```
Comment (aarshkshah1992) [inline introspect.go:168]: @raulk What is a "transport Id" ?

**Review by aarshkshah1992 — APPROVED** [review]
@raulk Beautiful Go :) LGTM. Just one question for my own curiosity & one comment to use `require.Eventually` in the test.

File: introspect_test.go [hunk 3]
```diff
@@ -32,58 +34,60 @@ func TestConnsAndStreamIntrospect(t *testing.T) {
...
-	// wait for the metres to kick in
+
+	// wait for the metrics to kick in
 	for {
-		cis, err := swarms[0].IntrospectConnections(introspect.ConnectionQueryParams{Output: introspect.QueryOutputFull})
+		cis, err := swarms[0].IntrospectConnections(introspection.ConnectionQueryParams{Output: introspection.QueryOutputFull})
 		require.NoError(t, err)
+
 		if cis[0].Traffic.TrafficOut.CumBytes != 0 {
 			break
```
Comment (aarshkshah1992) [inline introspect_test.go:44]: @raulk You might want to replace this with `require.Eventually`

Comment (raulk) [reply introspect_test.go:44]: @aarshkshah1992 yeah, feel free to submit a follow-up PR. I'm gonna go ahead and merge all of this now.

**Pull request closed without merging**